libraries/AP_BattMonitor/AP_BattMonitor_EFI.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_BattMonitor_EFI.h"
+
+#if HAL_EFI_ENABLED
+
+#include <AP_Common/AP_Common.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_EFI/AP_EFI.h>
+
+extern const AP_HAL::HAL& hal;
+
+// update state
+void AP_BattMonitor_EFI::read()
+{
+    AP_EFI *efi = AP::EFI();
```

Comment:
```suggestion
    const AP_EFI *efi = AP::EFI();
```

Reply:
get_state() is not const as it takes a semaphore